lib/puppet/file_system/file_impl.rb

```diff
@@ -54,14 +54,17 @@ def exclusive_open(path, mode, options = 'r', timeout = 300, &block)
     while !written
       ::File.open(path, options, mode) do |rf|
         if rf.flock(::File::LOCK_EX|::File::LOCK_NB)
+          Puppet.debug(_("Locked '%{path}'") % { path: path })
           yield rf
           written = true
+          Puppet.debug(_("Unlocked '%{path}'") % { path: path })
         else
+          Puppet.debug("Failed to lock '%s' retrying in %.2f milliseconds" % [path, wait * 1000])
           sleep wait
           timeout -= wait
           wait *= 2
           if timeout < 0
-            raise Timeout::Error, _("Timeout waiting for exclusive lock on %{path}") % { path: @path }
+            raise Timeout::Error, _("Timeout waiting for exclusive lock on %{path}") % { path: path }
```

Comment:
I'm curious what this looks like as actual output, does the `Timeout::Error` preface the string with something like "Timeout Error" ? If so saying Timeout again seems a little redundant. If not, I'd consider altering this to be "Timeout while waiting..." or "Timeout: Waiting for..."

Reply:
For this PR, the timeout message is never displayed, because the `Timeout::Error` exception is rescued and a LockError is raised instead. On the console you see:

```
$ bx puppet agent -t
Error: Could not run: Another puppet instance is already running; exiting
```

Reply:
If the `exclusive_open` fails, then the timeout error will be raised, and the filebucket save operation will fail, which is what I think we want.